docs/source/daml/intro/11_Testing.rst

```diff
@@ -0,0 +1,283 @@
+.. Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+11 Testing DAML Contracts
+=========================
+
+This chapter is all about testing and debugging the DAML contracts you've build using the tools from chapters 1-10. You've already met DAML Script as a way of testing your code inside the IDE. In this chapter you'll learn about other uses of DAML Script, as well as other tools you can use for testing and debugging. You'll also learn about a few error cases that are most likely to crop up only in actual distributed testing, and which need some care to avoid. Specifically we will cover:
+
+- DAML Test tooling - Script, REPL, and Navigator
+- The ``trace`` and ``debug`` functions
+- Contention
+
+Note that this section only covers testing your DAML contracts. For more holistic application testing, please refer to :doc:`/getting-started/testing`.
+
+If you no longer have your projects set up, please follow the setup instructions in :doc:`8_Dependencies` to get hold of the code for this chapter. There is no code specific to this chapter.
+
+DAML Test Tooling
+-----------------
+
+There are three primary tools available in the SDK to test and interact with DAML contracts:
+
+- DAML Script
+- DAML Navigator
+- DAML REPL
+
+DAML Script
+~~~~~~~~~~~
+
+:doc:`DAML Script </daml-script/index>` should be familiar by now. It's a way to script commands and queries from multiple parties against a DAML Ledger. Unless you've browsed other sections of the documentation already, you have probably used it mostly in the IDE. However, DAML Script can do much more than that. It has four different modes of operation:
+
+1. Run on a special Script Service in the IDE, providing the Script Views.
+2. Run the Script Service via the CLI, which is useful for quick regression testing.
+3. Start a Sandbox and run against that for regression testing against an actual Ledger API.
+4. Run against any other already running Ledger.
+
+Let's quickly see 2-4 in action. 
+
+Script Service CLI Testing
+..........................
+
+In your chapter 8 project, run ``daml test``. This runs the same Scripts that are run in the IDE on the same in-memory Ledger called the Script Service. Instead of views of the resulting ledger, it outputs successes and failures.
+
+.. code-block:: none
+
+  daml/Test/Intro/Asset/TradeSetup.daml:setupRoles: ok, 2 active contracts, 4 transactions.
+  daml/Test/Intro/Asset/TradeSetup.daml:test_issuance: ok, 3 active contracts, 5 transactions.
+  daml/Test/Intro/Asset/TradeSetup.daml:tradeSetup: ok, 6 active contracts, 10 transactions.
+  daml/Test/Intro/Asset/MultiTrade.daml:testMultiTrade: ok, 12 active contracts, 28 transactions.
+
+If you add the command line flag ``--junit``, ``daml test`` can write the results in jUnit format, which is useful for automated testing.
+
+Sandbox CLI Testing
```

Comment:
If you just want to ask for 4 usecases, the list above is enough and we can remove this section. If you want more then I think users should really just read the daml script docs. Our docs should primarily be there to help our users not to answer certification questions.